File: src/commands/importExportCommands.ts
// src/commands/importExportCommands.ts
import { core } from "@tauri-apps/api";
import type { ImportResult } from "../types";

export const importExportCommands = {
    async exportVault(): Promise<string> {
        return await core.invoke("export_vault");
    },

    async exportBoxAsEnv(
        boxId: string,
        prefix: string | null
    ): Promise<string> {
        return await core.invoke("export_box_as_env", { boxId, prefix });
    },

    async importVault(jsonData: string): Promise<ImportResult> {
        return await core.invoke("import_vault", { jsonData });
    },

    async importEnvToBox(
        envContent: string,
        targetBoxId: string
    ): Promise<ImportResult> {
        return await core.invoke("import_env_to_box", { envContent, targetBoxId });
    },
};

File: src-tauri/src/commands/import_export_commands.rs
// src/commands/import_export_commands.rs 
use crate::models::ImportResult;
use crate::states::AppState;
use std::sync::Arc;
use tauri::State;

#[tauri::command]
pub async fn export_vault(
    app_state: State<'_, Arc<AppState>>,
) -> Result<String, String> {
    app_state.export_vault().await.map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn export_box_as_env(
    box_id: String,
    prefix: Option<String>,
    app_state: State<'_, Arc<AppState>>,
) -> Result<String, String> {
    app_state
        .export_box_as_env(box_id, prefix)
        .await
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn import_vault(
    json_data: String,
    app_state: State<'_, Arc<AppState>>,
) -> Result<ImportResult, String> {
    app_state
        .import_vault(&json_data)
        .await
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn import_env_to_box(
    env_content: String,
    target_box_id: String,
    app_state: State<'_, Arc<AppState>>,
) -> Result<ImportResult, String> {
    app_state
        .import_env_to_box(&env_content, target_box_id)
        .await
        .map_err(|e| e.to_string())
}
